### PR TITLE
Locked travelnet compat

### PR DESCRIPTION
**Pull request by oversword**
I needed to make a few changes to make the new version of the mod compatible with `locked_travlenets`

These changes are not specific to `locked_travlenets` and should help with extending the travelnet mod in general.

None of these changes should change the current behaviour, just make it easier for people to modify the behaviour.

Changes:
* Get the pos from the context data when `travelnet.form_input_handler` is called, this allows someone to override `travelnet.on_receive_fields` without needing to have access to the pos data
* Allow the height of the form to be sent in as an option to the primary formspec, this lets you add another unit to make space for the locked travelnet specific form elements
* Call `travelnet.actions.end_input` every time the form is closed, this lets you override `travelnet.actions.end_input` as a way of hooking into the formspec closing event
* Re-arrange the way we get data for the update actions so that we can always pass in as much info as possible into `end_input`

### CODE REVIEW

**Comment by oversword** [comment]
These changes allow this MR to work https://github.com/BlockySurvival/locked_travelnet/pull/5/files?diff=unified&w=1

Which brings locked travelnets up to speed with the latest travelnet behaviour

File: on_receive_fields.lua [hunk 1]
```diff
@@ -116,7 +116,7 @@ function travelnet.on_receive_fields(pos, _, fields, player)
 	end
 
 	if not pos or not fields then
-		player_formspec_data[name] = nil
+		travelnet.actions.end_input(_, _, player)
```
Comment (S-S-X) [inline on_receive_fields.lua:119]: Underscore arguments should not be used, if you need argument give name for it and if you do not need then use nil.

Comment (oversword) [reply on_receive_fields.lua:119]: Good call actually, I should pass these in if they're available, hey might not be used now but there's no reason they couldn't be

Comment (oversword) [reply on_receive_fields.lua]: Passing in as much info as possible now

**Review by wsor4035 — APPROVED** [review]

**Pull request merged**